libzstd/encoder-standard/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+use core::slice;
+use std::io::Write;
+use std::os::raw::{c_char, c_uchar};
+use std::ptr::null;
+use zstd::stream::Encoder;
+use zstd::zstd_safe::{CParameter, ParamSwitch};
+
+// re-export zstd
+pub use zstd;
+
+// we use offset window no more than = 17
+// TODO: use for multi-block zstd.
+#[allow(dead_code)]
+pub const CL_WINDOW_LIMIT: usize = 17;
+
+/// zstd block size target.
+pub const N_BLOCK_SIZE_TARGET: u32 = 124 * 1024;
+
+/// Maximum number of blocks that we can expect in the encoded data.
+pub const N_MAX_BLOCKS: u64 = 10;
+
+/// Zstd encoder configuration
+pub fn init_zstd_encoder(target_block_size: u32) -> Encoder<'static, Vec<u8>> {
+    let mut encoder = Encoder::new(Vec::new(), 0).expect("infallible");
+
+    // disable compression of literals, i.e. literals will be raw bytes.
+    encoder
+        .set_parameter(CParameter::LiteralCompressionMode(ParamSwitch::Disable))
+        .expect("infallible");
+    // with a hack in zstd we can set window log <= 17 with single segment kept
+    encoder
+        .set_parameter(CParameter::WindowLog(17))
```

Comment:
For new encoder, increase the windowLog from 17 -> 22 (128K -> 4M), it would help to improve the compressing ratio and enable us a signle frame scheme for any source data less than 4MB (should be enough in quite a long peroid)

Reply:
Changed in this commit: [b64a7d3](https://github.com/scroll-tech/da-codec/pull/51/commits/b64a7d3cfc1046e8ae4fbd193b87fa2ba8348c84) and regenerated the `.a` files.